static/app/components/teams/createTeamForm.tsx

```diff
@@ -18,49 +17,41 @@ type Props = {
   formProps?: Partial<typeof Form>;
 };
 
-export default class CreateTeamForm extends Component<Props> {
-  handleSubmit = (data: Record<string, any>, onSuccess: Function, onError: Function) => {
-    callIfFunction(this.props.onSubmit, data as Payload, onSuccess, onError);
-  };
-
-  handleCreateTeamSuccess = (data: Payload) => {
-    callIfFunction(this.props.onSuccess, data);
-  };
-
-  render() {
-    const {organization} = this.props;
-
-    return (
-      <Fragment>
-        <p>
-          {t(
-            'Members of a team have access to specific areas, such as a new release or a new application feature.'
-          )}
-        </p>
-
-        <Form
-          submitLabel={t('Create Team')}
-          apiEndpoint={`/organizations/${organization.slug}/teams/`}
-          apiMethod="POST"
-          onSubmit={this.handleSubmit}
-          onSubmitSuccess={this.handleCreateTeamSuccess}
-          requireChanges
-          data-test-id="create-team-form"
-          {...this.props.formProps}
-        >
-          <TextField
-            name="slug"
-            label={t('Team Name')}
-            placeholder={t('e.g. operations, web-frontend, desktop')}
-            help={t('May contain lowercase letters, numbers, dashes and underscores.')}
-            required
-            stacked
-            flexibleControlStateSize
-            inline={false}
-            transformInput={slugify}
-          />
-        </Form>
-      </Fragment>
-    );
-  }
+function CreateTeamForm({organization, formProps, ...props}: Props) {
+  return (
+    <Fragment>
+      <p>
+        {t(
+          'Members of a team have access to specific areas, such as a new release or a new application feature.'
+        )}
+      </p>
+
+      <Form
+        submitLabel={t('Create Team')}
+        apiEndpoint={`/organizations/${organization.slug}/teams/`}
+        apiMethod="POST"
+        onSubmit={(data, onSuccess, onError) =>
```

Comment:
This is more of a question than a comment, but I see that we generally dont use a lot of the `useCallback` pattern in the codebase which means we probably waste some react cycles. E.g. here it would probably be helpful to use it because we are passing the callback as a prop and we could avoid unnecessary child rerenders. Is there a guideline for this today or would it make sense for us to add it? cc @matejminar here too, maybe I missed something 😅

Reply:
In this scenario, we could do that.
However, we are generally trying to not prematurely optimize as `useCallback` can sometimes cause more harm than good.
We usually fix these things opportunistically when we see something suspicious in performance monitoring.

Reply:
@matejminar true - it's easy to get it wrong. I've found that eslint `react-hooks/exhaustive-deps` is actually very good at finding those types of problems though. One nice benefit of useCallback is that it gently forces you to think of your function signatures a bit more and write code that's purer and easier to follow in order to avoid large dependency lists.

Reply:
yeah, `react-hooks/exhaustive-deps` is sweet, I also found this article from Kent very informative: https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
Any problem leaving this as it is for now?